src/datasets/containers.jl

```diff
@@ -46,4 +48,36 @@ isimagefile(file::String) = occursin(IMAGEFILE_REGEX, lowercase(file))
 const IMAGEFILE_REGEX = r"\.(gif|jpe?g|tiff?|png|webp|bmp)$"
 
 
-## TODO: TableDataset
+struct TableDataset{T}
+    table::T #Should implement Tables.jl interface
+    TableDataset{T}(table::T) where T = Tables.istable(table) ? new{T}(table) : error("Object doesn't implement Tables.jl interface")
+end
+
+TableDataset(path::String) = TableDataset(DataFrame(CSV.File(path)))
+TableDataset(table::T) where {T} = TableDataset{T}(table)
+
+function LearnBase.getobs(dataset::FastAI.Datasets.TableDataset{T}, idx) where {T}
+    if Tables.rowaccess(dataset.table)
+        for (index, row) in enumerate(Tables.rows(dataset.table))
+            if index==idx
+                return row
+            end
+        end
+    elseif Tables.columnaccess(dataset.table)
+        rowvals = []
+        for i in 1:length(Tables.columnnames(dataset.table))
+            append!(rowvals, Tables.getcolumn(dataset.table, i)[idx])
+        end
+        return rowvals
+    else error("The Tables.jl implementation used should have either rowaccess or columnaccess.")
+    end
+end
+
+function LearnBase.nobs(dataset::FastAI.Datasets.TableDataset{T}) where {T}
+    if Tables.rowaccess(dataset.table)
+        return length(Tables.rows(dataset.table))
```

Comment:
I saw a couple of implementations (e.g. JDBC) where `length` is not defined on the result of `rows`. The question is whether or not we care about that. There's also the question of whether this might end up materializing more data than we want as a side effect of calling `rows` or `columnaccess`. Unfortunately AFAICT there's no API in Tables.jl for querying the size of a source (if it has one).

Reply:
Then is the only option to iterate through the rows for the size, incase the table doesn't support column access? If it is, then should I make the required changes?